pages/careers/index.tsx

```diff
@@ -15,13 +16,40 @@ import Prose from "../../components/Prose/Prose";
 import clsx from "clsx";
 import { getHeadPageTitle } from "../../utils/meta";
 import { PageSettings } from "../../types/next";
+import {
+  getNewsEntries,
+  getCustomPageAndMicrocopy,
+} from "../../contentful/client";
+import {
+  TypeNews,
+  TypeMicrocopy,
+  TypeCustomPage,
+} from "../../generated/contentful";
+import Microcopy from "../../components/Microcopy/Microcopy";
 
 export const pageSettings: PageSettings = {
   path: "/careers",
   linkContent: <>Careers</>,
 };
 
-const Index = (): ReactElement => {
+export const getStaticProps: GetStaticProps<{
```

Comment:
this should be defined once and imported everywhere

Reply:
I like this idea, and just tried it. Nextjs threw the error I expected. Due to how nextjs effectively splits the codebase in two (client side and server side), I'm pretty sure getStaticProps *has* to be defined in the file that serves that page. When trying to have it reused and imported everywhere, nextjs became confused, thought it was client side code, and failed to render the pages.

Reply:
Maybe @lancegliser knows of a way to achieve this?